internal/brokerstore/brokerstore.go

```diff
@@ -0,0 +1,178 @@
+package brokerstore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/razorpay/metro/pkg/messagebroker"
+)
+
+// InstanceType ...
+type InstanceType string
+
+const (
+	producer InstanceType = "producer"
+	consumer InstanceType = "consumer"
+	admin    InstanceType = "admin"
+)
+
+// Key ...
+type Key struct {
+	instanceType InstanceType // producer, consumer, admin
+	topic        string       // topic name
+}
+
+// NewKey creates a new key for broker map
+func NewKey(instanceType, topic string) *Key {
+	return &Key{
+		instanceType: InstanceType(instanceType),
+		topic:        topic,
+	}
+}
+
+func (key *Key) String() string {
+	if key.instanceType == admin {
+		// admin clients are not associated with topic names
+		return fmt.Sprintf("%v", key.instanceType)
+	}
+	return fmt.Sprintf("%v-%v", key.instanceType, key.topic)
+}
+
+// BrokerStore holds broker clients
+type BrokerStore struct {
+	// stores active producer clients for a key.
+	producerMap map[*Key]messagebroker.Producer
+
+	// stores active producer clients for a key
+	consumerMap map[*Key]messagebroker.Consumer
```

Comment:
should these be sync maps?

Reply:
Was thinking of syncing using `consul lock` as `sync.Map` does not maintain types and accepts only `interface{}`. Anyways changed for now.

Reply:
we have seperate producer, consumer and admin maps, do we need to store type information in key as well ?

Reply:
That's to make the log readable. Else it will only be a topic name.